include/fixed_containers/optional_ref.hpp

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include "fixed_containers/optional_ref_checking.hpp"
+#include "fixed_containers/concepts.hpp"
+
+#include <functional>
+#include <optional>
+
+namespace fixed_containers
+{
+    // allows T& optionals
+    template <IsReference T,
+             customize::OptionalRefChecking<T> CheckingType = customize::OptionalRefAbortChecking<T>>
+    class OptionalRef
+    {
+        private:
+            using ref  = T; // T is required to be a reference type
+            using ptr = std::remove_reference<T>*;
+            using BackingType = std::optional<std::reference_wrapper<std::remove_reference_t<T>>>;
+            using Self = OptionalRef<T, CheckingType>;
+            BackingType underlying_type_; // younes todo makes this public so structural type?
```

Comment:
I think yes, note also that a barebones `StructuralTypeOptional` already exists in https://github.com/teslamotors/fixed-containers/blob/main/include/fixed_containers/enum_utils.hpp#L259C7-L259C29 , so an `Optional` type (not just references) might be useful in the future.